gobblin-api/src/main/java/gobblin/util/Decorator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util;
+
+import java.util.List;
+
+
+/**
+ * Interface for decorator patterns.
+ */
+public interface Decorator {
```

Comment:
My personal preference is that this interface just has a single method `getImmediateUnderlying()` or `getEnclosed()`. The two methods currently associated with it can be taken care of using static methods in DecoratorUtils. You can get another person's opinion.